src/client.rs

```diff
@@ -389,10 +389,14 @@ fn apply_despawns(
         // with the last replication message, but the server might not yet have received confirmation
         // from the client and could include the deletion in the this message.
         let server_entity = deserialize_entity(cursor)?;
-        if let Some(client_entity) = params
-            .entity_map
-            .remove_by_server(server_entity)
-            .and_then(|entity| world.get_entity_mut(entity))
+        if let Some(client_entity) =
+            params
+                .entity_map
+                .remove_by_server(server_entity)
+                .and_then(|entity| match world.get_entity_mut(entity) {
+                    Ok(entity) => Some(entity),
+                    Err(_) => None,
+                })
         {
```

Comment:
Let's simplify it a bit:
```suggestion
        if let Some(client_entity) = params
            .entity_map
            .remove_by_server(server_entity)
            .and_then(|entity| world.get_entity_mut(entity).ok())
```